.eslintrc.js

```diff
@@ -13,55 +23,92 @@ module.exports = {
     ecmaVersion: 11,
     sourceType: 'module',
   },
-  plugins: ['react', '@typescript-eslint', '@emotion', 'cypress'],
+  plugins: ['react', '@typescript-eslint', '@emotion', 'cypress', 'import'], // Add 'import' here
   rules: {
+    // General JavaScript/TypeScript
+    'no-console': 'warn',
+    'no-debugger': 'error',
+    'prefer-const': 'warn',
+    'no-var': 'error',
+    'consistent-return': 'warn',
+    'eqeqeq': ['error', 'always'],
+    '@typescript-eslint/explicit-module-boundary-types': 'warn',
+    '@typescript-eslint/explicit-function-return-type': 'warn',
+
+    // React Best Practices
+    'react/jsx-key': 'error',
+    'react/no-array-index-key': 'warn',
+    'react/jsx-curly-brace-presence': ['warn', { props: 'never', children: 'never' }],
+
+    // Code Quality
+    'complexity': ['warn', { max: 10 }],
+    'max-lines': ['warn', { max: 300, skipBlankLines: true, skipComments: true }],
+    'max-params': ['warn', MAX_PARAMS],
+    'no-magic-numbers': ['warn', { ignore: [0, 1] }],
+
+    // Security
+    'no-eval': 'error',
+    'no-new-func': 'error',
+
+    // Existing Rules
     'no-undef': 'warn',
     'no-unsafe-optional-chaining': 'warn',
     'no-use-before-define': 'off',
     '@typescript-eslint/no-use-before-define': 'warn',
     'no-empty-pattern': 'warn',
-    // base rule disabled as it can report incorrect errors
     'no-unused-vars': 'off',
     '@typescript-eslint/no-unused-vars': [
-      'warn', // or "error"
+      'warn', 
       {
         argsIgnorePattern: '^_',
         varsIgnorePattern: '^_',
         caughtErrorsIgnorePattern: '^_',
       },
     ],
-
     'react/react-in-jsx-scope': 'off',
-    'react/jsx-props-no-spreading': 'off',
+    'react/jsx-props-no-spreading': 'warn',
     'react/jsx-filename-extension': [
-      1,
-      { extensions: ['.js', '.jsx', '.tsx'] },
+      'warn', 
+      { extensions: ['.js', '.jsx', '.tsx'] }
     ],
     'react/no-unknown-property': [
-      2,
+      'error', 
       {
         ignore: ['jsx', 'global'],
-      },
+      }
+    ],
+    'react/jsx-indent': [
+      'warn', 
+      JSX_INDENT_SIZE
+    ],
+    'import/extensions': [
+      'warn', 
+      'always', 
+      {
+        js: 'never',
+        jsx: 'never',
+        ts: 'never',
+        tsx: 'never',
+      }
     ],
-    'react/jsx-indent': 'off',
-    'import/extensions': 'off',
-    'import/no-unresolved': 'off',
+    'import/no-unresolved': 'warn', // should be error
```

Comment:
These 2 rules seem to produce false positives - warnings are shown while importing js/jsx/ts/tsx files without an extension even though those are excluded.